src/reformat.c

```diff
@@ -1039,12 +1041,23 @@ static avifResult avifImageIdentity8ToRGB8ColorFullRange(const avifImage * image
         uint8_t * ptrG = &rgb->pixels[state->rgbOffsetBytesG + (j * rgb->rowBytes)];
         uint8_t * ptrB = &rgb->pixels[state->rgbOffsetBytesB + (j * rgb->rowBytes)];
 
-        for (uint32_t i = 0; i < image->width; ++i) {
-            avifStoreRGB8Pixel(rgb->format, ptrV[i], ptrY[i], ptrU[i], ptrR, ptrG, ptrB);
-
-            ptrR += rgbPixelBytes;
-            ptrG += rgbPixelBytes;
-            ptrB += rgbPixelBytes;
+        // This is intentionally a per-row conditional instead of a per-pixel
+        // conditional. This makes the "else" path (much more common than the
+        // "if" path) much faster than having a per-pixel branch.
+        if (rgb->format == AVIF_RGB_FORMAT_RGB_565) {
```

Comment:
Optional: Move this conditional outside the outer for loop.

Reply:
this would mean duplicating the outer for loop as well twice for no good reason. so i will leave this as-is.

Reply:
The `AVIF_RGB_FORMAT_RGB_565` version of the outer loop can omit the `ptrG` and `ptrB` variables. Since `rgb->format == AVIF_RGB_FORMAT_RGB_565` is a loop invariant, the compiler may make this this change as an optimization.